xarray/core/concat.py

```diff
@@ -586,9 +642,33 @@ def get_indexes(name):
                     )
                     result_vars[k] = v
             else:
-                combined_var = concat_vars(
-                    vars, dim, positions, combine_attrs=combine_attrs
-                )
+                # do not concat if:
+                # 1. variable is only present in one dataset of multiple datasets and
+                # 2. dim is not a dimension of variable
```

Comment:
I think you can already opt-in to this behaviour with `data_vars="minimal", coords="minimal"` in which case such variables are in `variables_to_merge`. Is there a test that will fail without this clause?

Reply:
I'll check this, thanks for this suggestion.

Reply:
https://github.com/pydata/xarray/blob/b741d8311fdc3d6150fd10d6421635eb4d84c7d5/xarray/tests/test_concat.py#L85-L91

This one breaks. But this test breaks also with current main, if we move `foo` to `data0`. So this test does only work because `foo` is not available in the first dataset.

Reply:
With these changes it works:

`actual = concat([data0, data1], "dim1", data_vars="minimal")`

Reply:
With https://github.com/pydata/xarray/pull/7400/commits/548ae19c3049663abe7cf2946b5dcdf5d47f35a3 this seems to work as expected.